src/devices/Rfid/Mifare/MifareCard.cs

```diff
@@ -0,0 +1,516 @@
+            {
+                C1 = 1;
+                C2 = 0;
+                C3 = 0;
+            }
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB | AccessType.WriteKeyB |
+                    AccessType.IncrementKeyB |
+                    AccessType.DecrementTransferRestoreKeyA | AccessType.DecrementTransferRestoreKeyB))
+            {
+                C1 = 1;
+                C2 = 1;
+                C3 = 0;
+            }
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB |
+                    AccessType.DecrementTransferRestoreKeyA | AccessType.DecrementTransferRestoreKeyB))
+            {
+                C1 = 0;
+                C2 = 0;
+                C3 = 1;
+            }
+
+            if (accessType == (AccessType.ReadKeyB | AccessType.WriteKeyB))
+            {
+                C1 = 0;
+                C2 = 1;
+                C3 = 1;
+            }
+
+            if (accessType == AccessType.ReadKeyB)
+            {
+                C1 = 1;
+                C2 = 0;
+                C3 = 1;
+            }
+
+            if (accessType == AccessType.Never)
+            {
+                C1 = 1;
+                C2 = 1;
+                C3 = 1;
+            }
+            // Encore the access bits
+            byte b6 = (byte)((((~C2) & 0x01) << (4 + blockNumber)) | (((~C1) & 0x01) << blockNumber));
+            byte b7 = (byte)(((C1) << (4 + blockNumber)) | (((~C3) & 0x01) << blockNumber));
+            byte b8 = (byte)(((C3) << (4 + blockNumber)) | ((C2) << blockNumber));
+            return (b6, b7, b8);
+        }
+
+        /// <summary>
+        /// Get the sector tailer access information
+        /// </summary>
+        /// <param name="blockNumber">the block sector number</param>
+        /// <param name="sectorData">The full sector data to decode</param>
+        /// <returns>the access sector rights</returns>
+        public AccessSector SectorTailerAccess(byte blockNumber, byte[] sectorData)
+        {
+            // Bit      7    6    5    4    3    2    1    0
+            // Byte 6 !C23 !C22 !C21 !C20 !C13 !C12 !C11 !C10 
+            // Byte 7  C13  C12  C11  C10 !C33 !C32 !C31 !C30
+            // Byte 8  C33  C32  C31  C30  C23  C22  C21  C20
+            // Cab a = access bit and b = block number
+            blockNumber = (byte)(blockNumber % 4);
+            if (blockNumber != 3)
+                return AccessSector.Never;
+
+            var (C1a, C1b, C2a, C2b, C3a, C3b) = DecodeSectorTailer(blockNumber, sectorData);
+            if (C1a != C1b)
+                return AccessSector.Never;
+
+            if (C2a != C2b)
+                return AccessSector.Never;
+
+            if (C3a != C3b)
+                return AccessSector.Never;
+
+            // Table of truth
+            if ((C1a == 0) && (C2a == 0) && (C3a == 0))
+                return AccessSector.WriteKeyAWithKeyA | AccessSector.ReadAccessBitsWithKeyA | AccessSector.ReadKeyBWithKeyA |
```

Comment:
is there any way to create those flags directly in such a way that all those ifs are not necessary?

Reply:
Unfortunately not. See documentation https://www.nxp.com/docs/en/data-sheet/MF1S70YYX_V1.pdf page 12 and 13

Reply:
a bit bummer design on the hardware :smile: